Node js Task2/RouteController.js

```diff
@@ -0,0 +1,108 @@
+const express = require("express");
+const router = express.Router();
+const users = require("./dataBase/data").users;
+const Joi = require('joi')
+
+const getUsers = (req, res) => {
+    res.send({ users });
+};
+const getUserById = (req, res) => {
+    const id = req.params.id;
+    const position = users.findIndex((user) => user.id === id);
+    if (position != -1) {
+        res.statusCode = 200;
+        res.send(users[position]);
+    } else {
+        res.statusCode = 404;
+        res.send({ message: "Unable to find the user" });
+    }
+};
+const UpdateUser = (req, res) => {
+    const id = req.params.id;
+    const data = req.body;
+    const position = users.findIndex((user) => user.id === id);
+    if (position != -1) {
+        const result = validateUser(data);
+        if (result.error == undefined) {
+            users[position] = data;
+            res.statusCode = 200;
+            res.send({ message: "Successfully Updated User's data" });
+        }
+        else {
+            res.statusCode = 404;
+            errorMessage = result.error.details;
+            res.send({ errorMessage });
+        }
+    } else {
+        res.statusCode = 404;
+        res.send({ message: "Unable to find the user" });
+    }
+};
+const getAutoSuggestUsers = (req,res)=>{
+    const loginSubstring = req.params.loginSubstring;
+    const limit = req.params.limit;
+
+    const SortedUsers = users.sort((a,b) => a.login.localeCompare(b.login));
+    const suggestedUsers = (SortedUsers.filter(user => user.login.indexOf(loginSubstring) != -1)).slice(0,limit);
+    res.statusCode = 200;
+    res.send({message:"The Suggested Users",Users:suggestedUsers});
+};
+const deleteUser = (req, res) => {
+    const id = req.params.id;
+    const position = users.findIndex((user) => user.id === id && user.isDeleted === false);
+    const userToBeDeleted = users[position];
+    if (position != -1) {
+        userToBeDeleted.isDeleted = true;
+        res.statusCode = 200;
+        res.send({ message: "Successfully Deleted a User's data" });
```

Comment:
Please correct cases under sentence for all the messages rest looks good

Reply:
change made